### PR TITLE
Always optimize `ClientBase::parseAndCheckOptions`

### PR DESCRIPTION
**Pull request by tavplubix**
### Changelog category (leave one):
- Not for changelog (changelog entry is not required)

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
See https://github.com/ClickHouse/ClickHouse/issues/65745. It doesn't solve the issue, but helps a bit 

### Documentation entry for user-facing changes

- [ ] Documentation is written (mandatory for new features)

> Information about CI checks: https://clickhouse.com/docs/en/development/continuous-integration/

#### CI Settings (Only check the boxes if you know what you are doing):
- [ ]  Allow: All Required Checks
- [x]  Allow: Stateless tests
- [x]  Allow: Stateful tests
- [ ]  Allow: Integration Tests
- [ ]  Allow: Performance tests
- [ ]  Allow: All Builds
- [ ]  Allow: batch 1, 2 for multi-batch jobs
- [ ]  Allow: batch 3, 4, 5, 6 for multi-batch jobs
---
- [ ]  Exclude: Style check
- [ ]  Exclude: Fast test
- [ ]  Exclude: All with ASAN
- [ ]  Exclude: All with TSAN, MSAN, UBSAN, Coverage
- [ ]  Exclude: All with aarch64, release, debug
---
- [ ]  Do not test
- [ ]  Woolen Wolfdog
- [ ]  Upload binaries for special builds
- [ ]  Disable merge-commit
- [ ]  Disable CI cache

### CODE REVIEW

**Comment by robot-ch-test-poll** [comment]
*This is an automated comment for commit 7d3cf887a843f5167512d222e7be2bec27676732 with description of existing statuses. It's updated for the latest CI running*

[✅ Click here](https://s3.amazonaws.com/clickhouse-test-reports/65746/7d3cf887a843f5167512d222e7be2bec27676732/ci_running.html) to open a full report in a separate page

<details><summary>Successful checks</summary>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>Style check</td><td>Runs a set of checks to keep the code style clean. If some of tests failed, see the related log from the report</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/65746/7d3cf887a843f5167512d222e7be2bec27676732/style_check.html">✅ success</a></td></tr>
<tbody>
</table>
</details>

**Review by alexey-milovidov — APPROVED** [review]

**Comment by alexey-milovidov** [comment]
Proper solution: - do not put settings from Settings.h into boost::program_options, but interpret them as custom options, similarly to `--param_xyz`

File: src/Client/ClientBase.cpp [hunk 1]
```diff
@@ -2862,7 +2862,10 @@ class OptionsAliasParser
 
 }
 
-
+/// Enable optimizations even in debug builds because otherwise options parsing becomes extremely slow affecting .sh tests
+#if defined(__clang__)
+#pragma clang optimize on
```
Comment (antonio2368) [inline src/Client/ClientBase.cpp:2867]: https://clang.llvm.org/docs/LanguageExtensions.html#extensions-for-selectively-disabling-optimization
Based on this, we cannot enable selectively parts to optimize?

Comment (tavplubix) [reply src/Client/ClientBase.cpp:2867]: Hmm, that's strange, I tested this locally, and it became slightly faster after adding this...

Comment (antonio2368) [reply src/Client/ClientBase.cpp:2867]: Seems like the same binary is generated with and without the tag

**Pull request merged**